handlers/handlers.go

```diff
@@ -1,238 +1,238 @@
+	responseWriter http.ResponseWriter, 
+	request *http.Request, 
+) { 
+	switch request.Method { 
+	case http.MethodGet: 
+		// Check to see if user is already logged in, if so redirect 
+		if _, err := getUserIdFromJwtToken(request); err == nil { 
+			http.Redirect( 
+				responseWriter, 
+				request, 
+				paths.HomePath, 
+				http.StatusTemporaryRedirect) 
+			return 
+		} 
+ 
+		parsedTemplate, err := template.ParseFiles("templates/login_or_signup.tmpl") 
+		if err != nil { 
+			log.Fatal(err) 
+		} 
+ 
+		parsedTemplate.Execute(responseWriter, nil) 
+ 
+	default: 
+		respondWithMethodNotAllowed(responseWriter, []string{http.MethodGet}) 
+	} 
+} 
+ 
+func HandleUserRequest( 
+	responseWriter http.ResponseWriter, 
+	request *http.Request, 
+) { 
+	type SignupForm struct { 
+		DisplayName  string `json:"displayName"` 
+		EmailAddress string `json:"emailAddress"` 
+		Password     string `json:"password"` 
+	} 
+ 
+	switch request.Method { 
+	case http.MethodPost: 
+		var signupForm SignupForm 
+		body := getRequestBody(request) 
+ 
+		if err := json.Unmarshal(body, &signupForm); err != nil { 
+			panic(err) 
+		} 
+ 
+		var statusCode int 
+		if err := userservice.StoreNewUser( 
+			signupForm.DisplayName, 
+			signupForm.EmailAddress, 
+			signupForm.Password, 
+		); err != nil { 
+			if err == userservice.EmailAddressAlreadyInUseError { 
+				statusCode = http.StatusConflict 
+			} else { 
+				panic(err) 
+			} 
+		} else { 
+			statusCode = http.StatusCreated 
+		} 
+ 
+		responseWriter.WriteHeader(statusCode) 
+ 
+	default: 
+		respondWithMethodNotAllowed(responseWriter, []string{http.MethodPost}) 
+	} 
+} 
+ 
+func HandleSessionRequest( 
+	responseWriter http.ResponseWriter, 
+	request *http.Request, 
+) { 
+	type LoginForm struct { 
+		EmailAddress string `json:"emailAddress"` 
+		Password     string `json:"password"` 
+	} 
+ 
+	switch request.Method { 
+	case http.MethodPost: 
+		var loginForm LoginForm 
+		body := getRequestBody(request) 
+ 
+		if err := json.Unmarshal(body, &loginForm); err != nil { 
+			panic(err) 
+		} 
+ 
+		if err := userservice.AuthenticateUserCredentials( 
+			loginForm.EmailAddress, 
+			loginForm.Password, 
+		); err != nil { 
+			panic(err) 
+		} 
+ 
+		// Set our cookie to have a valid JWT Token as the value 
+		{ 
+			userId, err := userservice.GetIdForUserWithEmailAddress(loginForm.EmailAddress) 
+			if err != nil { 
+				panic(err) 
+			} 
+ 
+			token, err := createTokenAsString(userId, credentialTimeoutInMinutes) 
+			if err != nil { 
+				panic(err) 
+			} 
+ 
+			expirationTime := time.Now().Add(credentialTimeoutInMinutes * time.Minute) 
+ 
+			cookie := http.Cookie{ 
+				Name:     cerealNotesCookieName, 
+				Value:    token, 
+				Expires:  expirationTime, 
+				HttpOnly: true, 
+			} 
+ 
+			http.SetCookie(responseWriter, &cookie) 
+		} 
+ 
+		responseWriter.WriteHeader(http.StatusCreated) 
+		responseWriter.Write([]byte(fmt.Sprint("passward email combo was correct"))) 
```

Comment:
`responseWriter` is a `io.Writer`, which means you can do
`fmt.Fprint(responseWriter, "password email combo was correct")`

there's also a `fmt.Fprintln` and `fmt.Fprintf` in case you ever need that